resources/aws_ec2.go

```diff
@@ -20,25 +20,46 @@ package resources
 import (
 	"context"
 	"encoding/base64"
+	"encoding/json"
 	"fmt"
+	"io/ioutil"
 	"log"
+	"net/http"
 	"sync"
 	"time"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/aws/aws-sdk-go/aws/request"
 	"github.com/aws/aws-sdk-go/aws/session"
+	cwe "github.com/aws/aws-sdk-go/service/cloudwatchevents"
 	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/sns"
+	multierr "github.com/hashicorp/go-multierror"
 	errwrap "github.com/pkg/errors"
 )
 
 func init() {
 	RegisterResource("aws:ec2", func() Res { return &AwsEc2Res{} })
 }
 
-// AwsPrefix is a const which gets prepended onto the instance name.
-const AwsPrefix = "mgmt:"
+const (
+	// AwsPrefix is a const which gets prepended onto the instance name.
+	AwsPrefix = "mgmt:"
+	// waitTimeout is the duration of the timeout context in CheckApply
+	waitTimeout = 400
+	// topicName is the name of the sns topic created by makeSnsTopic
+	topicName = "mgmtconfigEC2events"
+	// ruleName is the name of the cloudwatch topic created by makeCloudWatchRule
+	ruleName = "mgmtconfigEC2state"
+	// eventPattern describes the events the cloudwatch rule should detect
+	eventPattern = `{"source":["aws.ec2"],"detail-type":["EC2 Instance State-change Notification"],"detail":{"state":["running","stopped","terminated"]}}`
+	// snsPermission describes the security rule to allow cloudwatch to publish on sns
+	snsPermission = `{"Sid":"AllowSNSforCloudWatch","Effect":"Allow","Principal":{"Service":"events.amazonaws.com"},"Action":["SNS:Publish"],"Resource":"%s"}`
+	detail        = "$.detail"
```

Comment:
What are these? Add comments and/or clarify their var names. Probably don't need to call them `fooConst`, we know they're constant, specify the things we don't know. Eg: "snsProto" or similar.

Reply:
it originally was just sns, but that name is taken by the api. Will do.